src/python/module.cpp

```diff
@@ -1,10 +1,765 @@
+      .def("add_variables",
+           [](qc::QuantumComputation& qc,
+              const std::vector<qc::SymbolOrNumber>& vars) {
+             for (const auto& var : vars) {
+               qc.addVariable(var);
+             }
+           })
+      .def("is_variable_free", &qc::QuantumComputation::isVariableFree)
+      .def("get_variables", &qc::QuantumComputation::getVariables)
+      .def("initialize_io_mapping",
+           &qc::QuantumComputation::initializeIOMapping)
+      .def("from_file", nb::overload_cast<const std::string&>(
+                            &qc::QuantumComputation::import))
+      .def("from_file",
+           [](qc::QuantumComputation& qc, const std::string& filename,
+              const std::string& format) {
+             if (format == "qasm") {
+               qc.import(filename, qc::Format::OpenQASM);
+             } else if (format == "real") {
+               qc.import(filename, qc::Format::Real);
+             } else if (format == "grcs") {
+               qc.import(filename, qc::Format::GRCS);
+             } else if (format == "tfc") {
+               qc.import(filename, qc::Format::TFC);
+             } else if (format == "tensor") {
+               qc.import(filename, qc::Format::Tensor);
+             } else if (format == "qc") {
+               qc.import(filename, qc::Format::QC);
+             } else {
+               throw qc::QFRException("Unknown format: " + format);
+             }
+           })
+      .def("dump",
+           nb::overload_cast<const std::string&>(&qc::QuantumComputation::dump))
+      .def("dump",
+           [](qc::QuantumComputation& qc, const std::string& filename,
+              const std::string& format) {
+             if (format == "qasm") {
+               qc.dump(filename, qc::Format::OpenQASM);
+             } else if (format == "real") {
+               qc.dump(filename, qc::Format::Real);
+             } else if (format == "grcs") {
+               qc.dump(filename, qc::Format::GRCS);
+             } else if (format == "tfc") {
+               qc.dump(filename, qc::Format::TFC);
+             } else if (format == "tensor") {
+               qc.dump(filename, qc::Format::Tensor);
+             } else if (format == "qc") {
+               qc.dump(filename, qc::Format::QC);
+             } else {
+               throw qc::QFRException("Unknown format: " + format);
+             }
+           })
+      .def("to_open_qasm",
+           [](qc::QuantumComputation& qc) {
+             std::ostringstream oss;
+             qc.dumpOpenQASM(oss);
+             return oss.str();
+           })
+      .def("__len__", &qc::QuantumComputation::getNindividualOps)
+      .def(
+          "__getitem__",
+          [](const QuantumComputation& qc,
+             std::size_t
+                 idx) { return qc.at(idx).get(); }, // Beware: this gives write
+                                                    // access to underlying
```

Comment:
Do you mean the `at` method?

Reply:
Nah. That was one of the commented out code lines above or below.

In the lines here I meant to say that this comment should be in a docstring instead of in the source code.